### PR TITLE
Implement prefix tree week8 9

### PR DESCRIPTION
**Pull request by Giga-Chubinidze**

### CODE REVIEW

File: Trie/PrefixTree.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
+require_relative "PrefixTreeNode"
```
Comment (alexshvedko) [inline Trie/PrefixTree.rb]: PrefixTree.rb is camelcase, but we use snakecase in ruby

File: Trie/PrefixTree.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  end
+
+  def list(prefix = nil)
+    res = []
+    return root.data if prefix.nil?
+    return verified_words(container, prefix)
+  end
+
+  def verified_words(container, prefix)
+    res = []
```
Comment (alexshvedko) [inline Trie/PrefixTree.rb:57]: local variable is not used

File: Trie/PrefixTree.rb [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      return false unless current.data.include?(char)
+      current = current.data[char]
+    end
+    current.is_word
+  end
+
+  def list(prefix = nil)
+    res = []
+    return root.data if prefix.nil?
+    return verified_words(container, prefix)
```
Comment (alexshvedko) [inline Trie/PrefixTree.rb]: redundant `return`

File: Trie/PrefixTree.rb [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    current = root
+    prefix.each_char do |char|
+      return false unless current.data.include?(char)
+      current = current.data[char]
+    end
+    current.is_word
+  end
+
+  def list(prefix = nil)
+    res = []
```
Comment (alexshvedko) [inline Trie/PrefixTree.rb]: local variable is not used

File: Trie/PrefixTree.rb [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+trie = PrefixTree.new
+
+trie.add("qwerty")
+trie.add("qwcb")
+trie.add("cat")
+trie.add("dog")
+trie.add("doggoro")
+trie.add("dolphin")
+
+puts trie.list("do")
```
Comment (alexshvedko) [inline Trie/PrefixTree.rb]: trie.list() should return a list of words. In your case, it return tree

File: Trie/PrefixTree.rb (hunk 1 above)
Comment (Giga-Chubinidze) [reply Trie/PrefixTree.rb]: fixed

File: Trie/PrefixTree.rb (hunk 3 above)
Comment (Giga-Chubinidze) [reply Trie/PrefixTree.rb]: fixed

File: Trie/PrefixTree.rb (hunk 4 above)
Comment (Giga-Chubinidze) [reply Trie/PrefixTree.rb]: fixed

File: Trie/PrefixTree.rb (hunk 5 above)
Comment (Giga-Chubinidze) [reply Trie/PrefixTree.rb]: fixed

File: Trie/prefix_tree.rb [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+  def verified_words(container, prefix)
+    res = []
+    container.each do |word|
+      res << word if word.start_with?(prefix)
+    end
+    res.length == 0 ? false : res
+  end
+
+  def delete(word)
```
Comment (alexshvedko) [inline Trie/prefix_tree.rb:63]: method delete is not work
Screenshot 2022-05-04 at 12 50 11

Comment (Giga-Chubinidze) [reply Trie/prefix_tree.rb:63]: fixed

**Pull request merged**